verifier/verifier.go

```diff
@@ -43,90 +44,229 @@ import (
 	}
+
+	if ociTrustPolicy != nil {
+		if err := ociTrustPolicy.Validate(); err != nil {
+			return nil, err
+		}
+	}
+
+	if blobTrustPolicy != nil {
+		if err := blobTrustPolicy.Validate(); err != nil {
+			return nil, err
+		}
+	}
+
 	return &verifier{
-		trustPolicyDoc:   trustPolicy,
-		trustStore:       trustStore,
-		pluginManager:    pluginManager,
-		revocationClient: revocationClient,
+		ociTrustPolicyDoc:  ociTrustPolicy,
+		blobTrustPolicyDoc: blobTrustPolicy,
+		trustStore:         trustStore,
+		pluginManager:      pluginManager,
+		revocationClient:   revocationClient,
 	}, nil
+
+}
+
+// NewFromConfig returns a OCI verifier based on local file system
+// Deprecated: NewFromConfig function exists for historical compatibility and should not be used.
+// To create an OCI verifier, use NewOCIVerifierFromConfig function.
+func NewFromConfig() (notation.Verifier, error) {
+	return NewOCIVerifierFromConfig()
+}
+
+// New creates a new verifier given ociTrustPolicy, trustStore and pluginManager
+// Deprecated: New function exists for historical compatibility and should not be used.
+// To create verifier, use NewVerifier function.
+func New(ociTrustPolicy *trustpolicy.OCIDocument, trustStore truststore.X509TrustStore, pluginManager plugin.Manager) (notation.Verifier, error) {
+	return NewVerifier(ociTrustPolicy, nil, trustStore, pluginManager)
 }
 
 // SkipVerify validates whether the verification level is skip.
 func (v *verifier) SkipVerify(ctx context.Context, opts notation.VerifierVerifyOptions) (bool, *trustpolicy.VerificationLevel, error) {
 	logger := log.GetLogger(ctx)
 
 	logger.Debugf("Check verification level against artifact %v", opts.ArtifactReference)
-	trustPolicy, err := v.trustPolicyDoc.GetApplicableTrustPolicy(opts.ArtifactReference)
+	trustPolicy, err := v.ociTrustPolicyDoc.GetApplicableTrustPolicy(opts.ArtifactReference)
 	if err != nil {
 		return false, nil, notation.ErrorNoApplicableTrustPolicy{Msg: err.Error()}
 	}
 	logger.Infof("Trust policy configuration: %+v", trustPolicy)
+
 	// ignore the error since we already validated the policy document
 	verificationLevel, _ := trustPolicy.SignatureVerification.GetVerificationLevel()
-
 	// verificationLevel is skip
 	if reflect.DeepEqual(verificationLevel, trustpolicy.LevelSkip) {
 		logger.Debug("Skipping signature verification")
 		return true, trustpolicy.LevelSkip, nil
 	}
+
 	return false, verificationLevel, nil
 }
 
-// Verify verifies the signature blob `signature` against the target OCI
+// VerifyBlob verifies the signature of given blob , and returns the outcome upon
+// successful verification.
+func (v *verifier) VerifyBlob(ctx context.Context, descGenFunc notation.BlobDescriptorGenerator, signature []byte, opts notation.BlobVerifierVerifyOptions) (*notation.VerificationOutcome, error) {
+	logger := log.GetLogger(ctx)
+	logger.Debugf("Verify signature of media type %v", opts.SignatureMediaType)
+
+	var trustPolicy *trustpolicy.BlobTrustPolicy
```

Comment:
There should be a sanity check on `*trustpolicy.BlobTrustPolicy` for `VerifyBlob`. If `v.blobTrustPolicyDoc` is nil, the verification should fail immediately. This can happen because a user can use new an `OCI verifier`, but call `VerifyBlob` on it.